### PR TITLE
Adding sample service and other fixes.

### PR DESCRIPTION
**Pull request by arunachalams**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+## 1.9.6
+- Sample DropWizard App has been added in gojira-core, it serves both as a test and a sample application.
+- Refactored ParalleCallTest with lower concurrency, 25 threads and made it working.
+- Fixed a potential NPE in gojira-compare module.
+- Fixed potential bug with double-checked-locking: https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html
+- Renamed GojiraConstants -> GlobalConstants and added moved a few constants from TestStartEndTestHandler..
```
Comment (parthshyara) [inline CHANGELOG.md:6]: rewrite.

File: modules/gojira-core/src/test/java/com/flipkart/gojira/sample/app/SampleApp.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  @Override
+  public String getName() {
+    return "GojiraSample";
+  }
+
+  @Override
+  public void initialize(final Bootstrap<SampleAppConfiguration> bootstrap) {
+    // TODO: application initialization
```
Comment (parthshyara) [inline modules/gojira-core/src/test/java/com/flipkart/gojira/sample/app/SampleApp.java:46]: Is this TODO required?

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/injectors/GuiceInjector.java [hunk 3]
```diff
@@ -50,6 +51,23 @@ public static void assignInjector(Injector givenInjector) {
     }
   }
 
+  /**
+   * Mark assigned injector as null. Added for un-assigning when running tests.
+   */
+  @VisibleForTesting
+  public static void unAssignInjector() {
```
Comment (parthshyara) [inline modules/gojira-core/src/main/java/com/flipkart/gojira/core/injectors/GuiceInjector.java:58]: Can we define a graceful shutdowm mechanism for Gojira and make `unAssignInjector()` package protected?

Comment (arunachalams) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/injectors/GuiceInjector.java:58]: We have a mechanism for graceful shutdown via Managed. We are already shutting down connections as part of it. I have added TestQueuedSender shutdown as well. Anything other than these may be identified and added as part of another PR.

Comment (parthshyara) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/injectors/GuiceInjector.java:58]: Can we call the `unAssignInjector()` in Managed and call `Managed.shutdown()` in test cases? The main concern here is the public access to `unAssignInjector()`.

Comment (arunachalams) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/injectors/GuiceInjector.java:58]: I don't think "un-assigning injectors" is the right/correct way to do it and is not a standard practice AFAIK. @VisibleForTesting annotation is added here to capture the reason why it is added and should be fine IMO. It should not be part of Managed.shutdown().

**Pull request merged**